Sources/Fuzzilli/Compiler/Compiler.swift

```diff
@@ -479,6 +479,16 @@ public class JavaScriptCompiler {
             let value = try compileExpression(throwStatement.argument)
             emit(ThrowException(), withInputs: [value])
 
+        case .withStatement(let withStatement):
+            let object = try compileExpression(withStatement.object)
+            let beginWith = BeginWith()
+            emit(beginWith, withInputs: [object])
+            try enterNewScope {
+                for statement in withStatement.body.blockStatement.body {
```

Comment:
I think instead of this, you can just do `try compileBody(withStatement.body)`. Does that work?

Reply:
That's what I tried initially, however VSCode says: 

> For-in loop requires 'Compiler_Protobuf_Statement' to conform to 'Sequence'

Below is an example js code using the with-operator and the AST generated from the JS parser
# basic.js
```
var person = {
    firstName: "John",
  };
  
with (person) {
    console.log(firstName);
  }
  
```
# AST of basic.js
```
Fuzzilli.Compiler_Protobuf_AST:
statements {
  variableDeclaration {
    declarations {
      name: "person"
      value {
        objectExpression {
          fields {
            property {
              name: "firstName"
              value {
                stringLiteral {
                  value: "John"
                }
              }
            }
          }
        }
      }
    }
  }
}
statements {
  withStatement {
    object {
      identifier {
        name: "person"
      }
    }
    body {
      blockStatement {
        body {
          expressionStatement {
            expression {
              callExpression {
                callee {
                  memberExpression {
                    object {
                      identifier {
                        name: "console"
                      }
                    }
                    name: "log"
                  }
                }
                arguments {
                  identifier {
                    name: "firstName"
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```

Reply:
Hm which for-in loop does it complain about?
What I mean is something like
```swift
case .withStatement(let withStatement):
    let object = try compileExpression(withStatement.object)
    emit(BeginWith(), withInputs: [object])
    try enterNewScope {
        try compileBody(withStatement.body)
    }
    emit(EndWith())
```
`compileBody` requires only a `StatementNode`, so I think that should work there.

Reply:
I see. That works 👍